lib/widgets/compose_box.dart

```diff
@@ -378,32 +380,37 @@ class _FixedDestinationContentInput extends StatelessWidget {
     required this.narrow,
     required this.controller,
     required this.focusNode,
+    required this.enabled,
   });
 
   final SendableNarrow narrow;
   final ComposeContentController controller;
   final FocusNode focusNode;
+  final bool enabled;
 
   String _hintText(BuildContext context) {
     final zulipLocalizations = ZulipLocalizations.of(context);
-    switch (narrow) {
-      case TopicNarrow(:final streamId, :final topic):
+    switch ((narrow, enabled)) {
+      case (TopicNarrow(:final streamId, :final topic), _):
         final store = PerAccountStoreWidget.of(context);
         final streamName = store.streams[streamId]?.name
           ?? zulipLocalizations.composeBoxUnknownChannelName;
         return zulipLocalizations.composeBoxChannelContentHint(streamName, topic);
 
-      case DmNarrow(otherRecipientIds: []): // The self-1:1 thread.
+      case (DmNarrow(otherRecipientIds: []), _): // The self-1:1 thread.
         return zulipLocalizations.composeBoxSelfDmContentHint;
 
-      case DmNarrow(otherRecipientIds: [final otherUserId]):
+      case (DmNarrow(otherRecipientIds: [final otherUserId]), true):
         final store = PerAccountStoreWidget.of(context);
         final fullName = store.users[otherUserId]?.fullName;
         if (fullName == null) return zulipLocalizations.composeBoxGenericContentHint;
         return zulipLocalizations.composeBoxDmContentHint(fullName);
 
-      case DmNarrow(): // A group DM thread.
+      case (DmNarrow(), true): // A group DM thread.
         return zulipLocalizations.composeBoxGroupDmContentHint;
+
+      case (DmNarrow(), false):
+        return zulipLocalizations.composeBoxDeactivatedDmContentHint;
```

Comment:
Here we're relying on an assumption that if `enabled` is false, it's because there's a deactivated user.

That's how the code currently works in this version, as of the tip of this PR. But it's not expressed in the name `enabled`, nor in any dartdoc on that field. So now that the field is added, it'd be very natural for someone in the future to add some other condition that causes `enabled` to be false; but that would produce a misleading message here.

One adequate fix would be to give the field a more specific name that matches the semantics this widget is attaching to it. For example, `disabledBecauseDmRecipientDeactivated`. Then this code can look for that flag, and return this specific error message when it's set, with a clear conscience.

A more complex fix would be to make an enum in the same spirit as `TopicValidationError` and `ContentValidationError`, with a name like `DestinationError`. It'd have just one value at first, named like `DestinationError.dmRecipientDeactivated`. Then the flag here would be declared like
```dart
  final DestinationError? destinationError;
```
and null would mean there was no error.

The advantage of that more complex fix is that it'd be straightforward to add another value to the enum in the future, for a different error condition (like running afoul of the stream post policy, #674).